models/user_input.js

```diff
@@ -0,0 +1,174 @@
+                            'description, timestamp, created_at, updated_at) ' +
+                            'VALUES ($1, $2, $3, $4, $5, $6, $6) returning experiment_id, device_id',
+                            [data.experiment_id, data.device_id, data.user_id,
+                            data.description, data.timestamp, time]);
+                    })
+                    .then((result) => {
+                        resolve(result.rows[0]);
+                    })
+                    .catch((err) => {
+                        console.log(err);
+                        reject(err);
+                    });
+            });
+        },
+
+        delete: (data) => {
+            var time = new Date().getTime();
+            return new Promise((resolve, reject) => {
+                _db.query('UPDATE user_inputs SET deleted_at = $2 WHERE id = $1 returning id', [data.id, time])
+                    .then((result) => {
+                        resolve(result.rows[0]);
+                    })
+                    .catch((err) => {
+                        reject(err);
+                    });
+            });
+        },
+
+        updateDescription: (data) => {
+            var time = new Date().getTime();
+            return new Promise((resolve, reject) => { 
+                _db.query('UPDATE user_inputs SET description = $2, updated_at = $3 WHERE id = $1 and deleted_at = 0 returning description',
+                    [data.id, data.description, time])
+                    .then((result) => {
+                        resolve(result.rows[0]);
+                    })
+                    .catch((err) => {
+                        console.log(err);
+                        reject(err);
+                    });
+            });
+        },
+    }
+};
+
+function findOneById(id) {
+    return new Promise((resolve, reject) => {
+      _db.query('SELECT * FROM user_inputs WHERE id = $1 and deleted_at = 0', [id])
+        .then((result) => {
+          if (result.rows[0]) {
+            resolve(result.rows[0]);
+          }
+          else {
+            reject('no user input found')
+          }
+        })
+        .catch((err) => {
+          reject(err);
+        });
+    });
+}
+
+function validateUserInputData(data) {
+    return new Promise((resolve, reject) => {
+        if (!data.description) {
+            reject('invalid input: description missing')
+        }
+        else{
+            columns = ['user_id','device_id', 'description', 'experiment_id']
+            _Validator.validateColumns(data, columns)
+            .then(function () {
+                return _Validator.validateUserId(data.device_id)
+            })
+            .then(function () {
+                return _Validator.validateUserId(data.user_id)
+            })
+            .then(function() {
+                return _Validator.validateExperimentId(data.experiment_id)
+            })
+            .then((description) => {
+                description = data.description;
```

Comment:
What is this aiming to do here? 
2 issues I see:
- `description` is going to be an object representing the experiment so it shouldn't be called description
- setting `description = data.description` isn't going to do anything as the `description` variable doesn't have anything done with it after it is updated.

Reply:
When I had it as, `data.description = description`, it was setting the description of the experiment to be the description of the `user_input`. I'm not completely sure how, but it seems that somewhere in the method, the context is changed to be from the experiment's data rather than that of the user input. 

I can show you at our next meeting and maybe we can do some digging to figure out what exactly is happening to cause that behavior

Reply:
@Jss7268